src/implementation.ts

```diff
@@ -179,23 +186,44 @@ function getBatchLoader(resourceConfig: BatchResourceConfig, resourcePath: Reado
                 const requests = requestIDs.map(id => keys[id]);
 
                 ${(() => {
-                    const { batchKey, newKey, commaSeparatedBatchKey } = resourceConfig;
+                    const {
+                        batchKey,
+                        newKey,
+                        secondaryBatchKey,
+                        secondaryNewKey,
+                        commaSeparatedBatchKey,
+                    } = resourceConfig;
 
                     let batchKeyParam = `['${batchKey}']: requests.map(k => k['${newKey}'])`;
                     if (commaSeparatedBatchKey === true) {
                         batchKeyParam = `${batchKeyParam}.join(',')`;
                     }
 
-                    return `
-                        // For now, we assume that the dataloader key should be the first argument to the resource
-                        // @see https://github.com/Yelp/dataloader-codegen/issues/56
-                        const resourceArgs = [{
-                            ..._.omit(requests[0], '${resourceConfig.newKey}'),
-                            ${batchKeyParam},
-                        }];
-                    `;
+                    if (secondaryNewKey !== undefined && secondaryBatchKey !== undefined) {
```

Comment:
https://blog.larah.me/write-explicit-type-guards/

What if `secondaryNewKey` is somehow `null`? :P